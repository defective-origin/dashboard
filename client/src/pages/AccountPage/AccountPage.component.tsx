import React from 'react'

// ---| core |---
import { cn } from 'tools'
import { Outlet } from 'router'

// ---| pages |---
import Page, { PageMenuItem, PageProps } from 'pages/Page'

// ---| screens |---
// ---| components |---
// ---| self |---
import css from './AccountPage.module.scss'

export type AccountPageProps = PageProps

/**
 * Component description.
 *
 * How to use
 * @example
 * <AccountPage />
 */
export function AccountPage(props: AccountPageProps): JSX.Element {
  const { children, className, ...otherProps } = props
  const _className = cn(css.AccountPage, className)
  const menu: PageMenuItem[] = [
    { variant: 'nav', content: 'Account', to: 'ACCOUNT' },
    { variant: 'nav', content: 'Widgets', to: 'ACCOUNT_WIDGETS' },
    { variant: 'nav', content: 'Dashboards', to: 'ACCOUNT_BOARDS' },
  ]

  return (
    <Page className={_className} name='PAGES.ACCOUNT' menu={menu} {...otherProps}>
      <Page.Content p='xs'>
        <Outlet />

        {children}
      </Page.Content>
    </Page>
  )
}

AccountPage.displayName = 'AccountPage'

export default AccountPage
