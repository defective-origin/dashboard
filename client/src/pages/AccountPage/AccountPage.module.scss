@import 'src/assets/styles';

:root {
  // --ui-framework-color: var(--account-framework-color);
}

.AccountPage {
}
