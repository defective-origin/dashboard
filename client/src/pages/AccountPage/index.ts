/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './AccountPage.component'

/* INJECT_EXPORT_PLACE */
export * from './AccountPage.component'
