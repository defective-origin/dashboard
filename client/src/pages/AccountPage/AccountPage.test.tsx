// ---| tests |---
import { render } from 'tests'

// ---| self |---
import AccountPage from './AccountPage.component'

describe('[AccountPage] component', () => {
  it('should render component', () => {
    const container = render(<AccountPage />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
