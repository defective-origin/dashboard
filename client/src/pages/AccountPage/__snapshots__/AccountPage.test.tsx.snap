// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[AccountPage] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="_AccountPage_4206e2"
  />
</DocumentFragment>
`;
